Try to fix #19803

monojenkins opened the pull request:
!! This PR is a copy of mono/mono#19876,  please do not edit or review it in this repo !!<br/>Do not automatically approve this PR:<br/><br/>* Consider how the changes affect configurations in this repo,<br/>* Check effects on files that are not mirrored,<br/>* Identify test cases that may be needed in this repo.<br/><br/>!! Merge the PR only after the original PR is merged !!<br/><hr/><br/>
My previous unsuccessful pull request   https://github.com/mono/mono/pull/19804
I have improved the logic so that the code is more systematic

alexger97 commented:
Hello! Finally, the tests give a positive result. I think that this code can be merge

alexger97 committed:
Try to fix mono/mono#19803

My previous unsuccessful pull request   https://github.com/mono/mono/pull/19804
I have improved the logic so that the code is more systematic

danmosemsft commented:
@lambdageek is this PR still relevant? It is very old now.

danmosemsft commented:
ping @lambdageek
